core/src/test/java/org/apache/iceberg/hadoop/TestHadoopTables.java

```diff
@@ -139,6 +139,18 @@ public void testCustomSortOrder() {
     Assert.assertEquals("Transform must match", transform, sortOrder.fields().get(0).transform());
   }
 
+  @Test
+  public void testTableName() {
+    PartitionSpec spec = PartitionSpec.builderFor(SCHEMA)
+        .bucket("data", 16)
+        .build();
+    String location = tableDir.toURI().toString();
+    TABLES.create(SCHEMA, spec, location);
+
+    Table table = TABLES.load(location);
```

Comment:
Metadata tables will have their names as location.type (which is weird as we normally use location#type). I am not sure whether it is a big deal or not.

Reply:
I think ideally you would be able to use the result of `name` to load the table. I'd prefer to fix it if we can. Maybe we should pass the full metadata table name in, rather than having it added by the table implementation.

Reply:
I fixed it but it did require substantially more changes.